### PR TITLE
Earlier incumbent selection for successive halving

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: smac/intensification/successive_halving.py [hunk 1]
```diff
@@ -291,7 +294,7 @@ def eval_challenger(self,
         #   - this check is required because there is no incumbent performance
         #     for the first ever 'intensify' run (from initial design)
         if incumbent:
-            inc_runs = run_history.get_runs_for_config(incumbent)
+            inc_runs = run_history.get_runs_for_config(incumbent, only_max_observed_budget=True)
```
Comment (KEggensperger) [inline smac/intensification/successive_halving.py]: Check why line 304 is different from line 421

File: smac/intensification/successive_halving.py [hunk 2]
```diff
@@ -550,11 +555,61 @@ def _get_incumbent(self,
...
+                        else:
+                            self.logger.debug("  %s remains unchanged: %r" %
+                                              (param[0], param[1]))
+                    new_incumbent = challenger
+                else:
+                    self.logger.debug("Incumbent (%.4f) is at least as good as the challenger (%.4f) on budget %.4f.",
+                                      inc_cost, chall_cost, inc_run.budget)
+                    new_incumbent = incumbent
+
+                print(inc_cost, chall_cost)
```
Comment (KEggensperger) [inline smac/intensification/successive_halving.py]: Do we need this print?

File: smac/intensification/successive_halving.py [hunk 3]
```diff
@@ -550,11 +555,61 @@ def _get_incumbent(self,
...
+        else:
```
Comment (KEggensperger) [inline smac/intensification/successive_halving.py]: This is unreachable code?

File: smac/intensification/successive_halving.py (hunk 2 above)
Comment (mfeurer) [reply smac/intensification/successive_halving.py]: No, but I already removed it.

File: smac/intensification/successive_halving.py (hunk 3 above)
Comment (mfeurer) [reply smac/intensification/successive_halving.py]: I will adapt the code.

File: smac/intensification/successive_halving.py (hunk 1 above)
Comment (mfeurer) [reply smac/intensification/successive_halving.py]: I simplified the code to remove the confusion here.

**Pull request merged**